### DISCUSSION TITLE
Add second parameter to responseheader

### DISCUSSION THREAD

**Discussion by mtxadmin** [uBlockOrigin/uBlock-issues#2727] [Ideas]
According to [wiki](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#response-header-filtering), example.com##^responseheader(set-cookie)
can prevent all cookies from example.com to reach the browser. But usually it's more useful to prevent only some "bad" cookies, leaving other "good" cookies intact.

Yes, there is a [construction](https://github.com/uBlockOrigin/uBlock-issues/wiki/Resources-Library#cookie-removerjs-) host##+js(cookie-remover, filter) to counter tracking cookies, and it can use regexps. It works pretty [good](https://github.com/mtxadmin/ublock/blob/master/filters/_cookies), But if server sets a tracking cookie with httpOnly flag, this js scriptlet cannot counter it.

I am asking to add second parameter to responseheader, with a name of cookie, or a regexp filter like in cookie-remover.js

**Comment by krystian3w** [comment]
AdGuard's syntax should support block create by name:

https://adguard.com/kb/general/ad-filtering/create-own-filters/#cookie-modifier

> 
> ```
> $cookie [= name[; maxAge = seconds [; sameSite = strategy ]]]
> ```
>

If you prefere fix by older idea than his `$header`:

https://adguard.com/kb/general/ad-filtering/create-own-filters/#header-modifier

> 
> -   `||example.com^$header=set-cookie:foo` blocks requests whose responses have the Set-Cookie header with the value matching foo literally.
> -   `||example.com^$header=set-cookie blocks` requests whose responses have the Set-Cookie header with any value.
> -   `@@||example.com^$header=set-cookie:/foo\, bar\$/` unblocks requests whose responses have the Set-Cookie header with value matching the foo, bar$ regular expression.
> -   `@@||example.com^$header=set-cookie` unblocks requests whose responses have a Set-Cookie header with any value.
> 

Their wiki indicates that even the Firefox add-on does not support it `$header`. You can judge how much their header differs from this experimental one: https://github.com/gorhill/uBlock/wiki/Static-filter-syntax#header
